internal/tableland/impl/mesa.go

```diff
@@ -49,6 +49,37 @@ func (t *TablelandMesa) ValidateCreateTable(
 	return tableland.ValidateCreateTableResponse{StructureHash: createStmt.GetStructureHash()}, nil
 }
 
+// ValidateWriteQuery allows the user to validate a write query.
+func (t *TablelandMesa) ValidateWriteQuery(
+	ctx context.Context,
+	req tableland.ValidateWriteQueryRequest) (tableland.ValidateWriteQueryResponse, error) {
+	ctxController := ctx.Value(middlewares.ContextKeyAddress)
+	controller, ok := ctxController.(string)
+	if !ok || controller == "" {
+		return tableland.ValidateWriteQueryResponse{}, errors.New("no controller address found in context")
+	}
+	ctxChainID := ctx.Value(middlewares.ContextKeyChainID)
+	chainID, ok := ctxChainID.(tableland.ChainID)
+	if !ok {
+		return tableland.ValidateWriteQueryResponse{}, errors.New("no chain id found in context")
+	}
+	_, chainOk := t.chainStacks[chainID]
+	if !chainOk {
+		return tableland.ValidateWriteQueryResponse{}, fmt.Errorf("chain id %d isn't supported in the validator", chainID)
+	}
+
+	mutatingStmts, err := t.parser.ValidateMutatingQuery(req.Statement, chainID)
```

Comment:
`t.parser.ValidateMutatingQuery` does not appear to check if the table exists.
I wrote an e2e test [here](https://github.com/tablelandnetwork/local-tableland/blob/joe/norelay-writes/test/test.js#L185) that will call this with a table name that doesn't exist. The call to the sdk's `write`does not fail as expected.
It seems good if we can prevent the client from paying for a statement that will fail if it isn't too difficult on our end.
@jsign would there be a way to check the tablename's existence here without too much overhead?
An alternative is we could respond from this request with the tablename in addition to the tableID, then the SDK could do a second request for a read from the table, e.g. `select * from ${tablename} limit 1;`

Reply:
Correct, `ValidateMutatingQuery` is a feature at the parsing level, so it's very light and fast to run.

> @jsign would there be a way to check the tablename's existence here without too much overhead?

Yes, we could change L66 to `stack , chainOk := t.chainStacks[chainID]`, and then do `_, err := stack.GetTable(ctx, tableID)`. Today, that `GetTable` would fail if that table doesn't exist but we don't have any "sentinel error" to do a concrete check if the failure is for that reason. If we add that, we could do:
```
if _, err := stack.GetTable(ctx, tableID); err == ErrTableNotFound {
   return tablleand.validateWriteQueryResponse{}, errrors.New("table doesn't exist")
}
```
The `ErrTableNotFound` is an error that doesn't exist yet. If you want, I can add that to your branch here.

Maybe might be worth adding an extra flag in the request body `OnlyParse bool` or similar, if somebody only wants to do syntax parsing? I'm not sure about this, so mostly thinking out loud. I'm not sure if anyone would like to prepare and check write-query statements before they actually create the table. If the usual use case is usually assuming the table should exist, the `OnlyParse` default value would be `false`. So this other path would be mostly optional.